x/feegrant/filtered_fee_test.go

```diff
@@ -0,0 +1,208 @@
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+				Expiration: &oneHour,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       smallAtom,
+			blockTime: now,
+			accept:    true,
+			remove:    false,
+			remains:   leftAtom,
+		},
+		"expired": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+				Expiration: &now,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       smallAtom,
+			blockTime: oneHour,
+			accept:    false,
+			remove:    true,
+		},
+		"fee more than allowed": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+				Expiration: &oneHour,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       bigAtom,
+			blockTime: now,
+			accept:    false,
+		},
+		"with out spend limit": {
+			allowance: &feegrant.BasicAllowance{
+				Expiration: &oneHour,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       bigAtom,
+			blockTime: now,
+			accept:    true,
+		},
+		"expired no spend limit": {
+			allowance: &feegrant.BasicAllowance{
+				Expiration: &now,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       bigAtom,
+			blockTime: oneHour,
+			accept:    false,
+		},
+	}
+
+	for name, stc := range cases {
+		tc := stc // to make scopelint happy
+		t.Run(name, func(t *testing.T) {
+			err := tc.allowance.ValidateBasic()
+			require.NoError(t, err)
+
+			ctx := app.BaseApp.NewContext(false, ocproto.Header{}).WithBlockTime(tc.blockTime)
+
+			// create grant
+			createGrant := func() feegrant.Grant {
+				var granter, grantee sdk.AccAddress
+				allowance, err := feegrant.NewAllowedMsgAllowance(tc.allowance, tc.msgs)
+				require.NoError(t, err)
+				grant, err := feegrant.NewGrant(granter, grantee, allowance)
+				require.NoError(t, err)
+				return grant
+			}
+			grant := createGrant()
+
+			// create some msg
+			call := feegrant.MsgRevokeAllowance{
+				Granter: "",
+				Grantee: "",
+			}
+
+			// now try to deduct
+			allowance, err := grant.GetGrant()
+			require.NoError(t, err)
+			removed, err := allowance.Accept(ctx, tc.fee, []sdk.Msg{&call})
+			if !tc.accept {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+
+			require.Equal(t, tc.remove, removed)
+			if !removed {
+				updatedGrant := func(granter, grantee sdk.AccAddress,
+					allowance feegrant.FeeAllowanceI) feegrant.Grant {
+					newGrant, err := feegrant.NewGrant(
+						granter,
+						grantee,
+						allowance)
+					require.NoError(t, err)
+
+					cdc := simapp.MakeTestEncodingConfig().Codec
+					bz, err := cdc.Marshal(&newGrant)
+					require.NoError(t, err)
+
+					var loaded feegrant.Grant
+					err = cdc.Unmarshal(bz, &loaded)
+					require.NoError(t, err)
+					return loaded
+				}
```

Comment:
Mimic the logic as best as I can.
After `Accept()` has been called, the updated grant should be saved, so I added `Marshal()`.
To check the values are all correct, we must load the grant again, so I added `Unmarshal()`.